src/Hooks/Database/Support/QueryManager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the tenancy/tenancy package.
+ *
+ * Copyright Tenancy for Laravel
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://tenancy.dev
+ * @see https://github.com/tenancy
+ */
+
+namespace Tenancy\Hooks\Database\Support;
+
+use Closure;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Database\QueryException;
+
+class QueryManager
```

Comment:
Also, let's make this macroable, just in case :)

Reply:
I can certainly do that, but what do you expect to macro here?

Reply:
What if I want the ability in the closure to set / change the connection?
Or what if I would like to do X or Y. Macroable simply helps with that without having to extend things.